### DISCUSSION TITLE
JSQLParser very slow in parsing long queries

### DISCUSSION THREAD

**Discussion by salonijuneja** [JSQLParser/JSqlParser#1296] [General]
We have got queries that are 151k characters long. JSQLParser is slowing down and not returning response for more than half hour now. Is it a known issue? Will increasing jvm memory configuration help resolve it?

Sorry but I cannot share the official query. Please let me know if it is a known issue.

**Comment by manticore-projects** [comment]
Greetings.

The size of the query (in characters length) does not matter much.
What matters most is **the nesting of expressions**.

Also you should double check which version you are using as 4.1 and latest 4.2-Snapshots brought massive improvements in this regard.

Last but not least, check what kind of features is used in the query. Everything like Oracle 11.2 would not be an issue.
But later than that, some syntax have been introduced which are not supported yet. Of course these would fail eventually, but before they fail they can put the JSQLParser on a wrong trail (e.g. starting deep nested parsing of complex expressions).

Best regards
Andreas

**Comment by salonijuneja** [comment]
Hi Andreas,

Thanks for the insights. When is version 4.2 releasing? We use maven only. I can probably test out after release.

**Comment by manticore-projects** [comment]
On Tue, 2021-08-03 at 12:40 +0000, salonijuneja wrote:
> Thanks for the insights. When is version 4.2 releasing? We use maven
> only. I can probably test out after release.

This is out of my hands, I am just the janitor.
Although you can use the SNAPSHOT repo with Maven:

<repositories>
     <repository>
         <id>jsqlparser-snapshots</id>
         <snapshots>
             <enabled>true</enabled>
         </snapshots>
         <url>https://oss.sonatype.org/content/groups/public/</url>
     </repository>
</repositories>

Best regards
Andreas

**Comment by salonijuneja** [comment]
Thanks. But we cannot use snapshots from sonatype :(. I will wait for the next release!

**Comment by manticore-projects** [comment]
Please retry with released JSQLParser-8.0 which saw some material performance improvements (like real life queries from 44 seconds down to 2 seconds)
